Jumpscale/clients/blockchain/tfchain/TFChainReservation.py

```diff
@@ -0,0 +1,105 @@
+from Jumpscale import j
+import nacl
+
+_RESERVATION_SCHEMA = j.data.schema.get("""
+@url = tfchain.reservation
+type = "S3,VM" (E)
+duration = (I)
+size = (I)
+email = (email)
+bot_id = (S)
+created = (I)
+""")
+
+_GRID_BROKER_PUB_KEY = "..."
+
+
+class TFChainReservation():
+    """
+    TFChainReservation contains all TF grid reservation logic.
+    """
+
+    def __init__(self, wallet):
+        self._wallet = wallet
+        self._notary_client = None
+        self._grid_broker_pub_key = None
+        self._grid_broker_addr = ''  # TODO: hardcode tfchain address of the grid broker
+        # this all key management needs to be improved
+        keypair = self._wallet.key_pair_get(self._wallet.address)
+        self._signing_key = nacl.signing.SigningKey(keypair.private_key.to_seed())
+        self._private_key = self._signing_key.to_curve25519_private_key()
+
+    @property
+    def notary_client(self):
+        if self._notary_client is None:
+            self._notary_client = j.clients.gedis.configure('tfnotary', host='notary.grid.tf', port=8888)
+        return self._notary_client
+
+    @property
+    def grid_broker_pub_key(self):
+        if self._grid_broker_pub_key is None:
+            record = self._wallet.client.threebot.record_get('tf3bot.zaibon')
+            encoded_key = record.public_key.unlockhash
+            self._grid_broker_pub_key = nacl.public.PublicKey(
+                str(encoded_key.hash),
+                encoder=nacl.encoding.HexEncoder)
+        return self._grid_broker_pub_key
+
+    def new(self):
+        return _RESERVATION_SCHEMA.new()
+
+    def send(self, reservation):
+        reservation.created = j.data.time.epoch
```

Comment:
I usually do a type check here to give more clear errors should a wrong type be given

Reply:
I'll add it once I'm settle on which type is going to be accepted there. This goes in pair with my previous comment